src/ICS20Transfer.sol

```diff
@@ -78,16 +81,32 @@ contract ICS20Transfer is IIBCApp, IICS20Transfer, IICS20Errors, Ownable, Reentr
             revert ICS20MsgSenderIsNotPacketSender(msg_.sender, sender);
         }
 
-        // transfer the tokens to us (requires the allowance to be set)
-        _transferFrom(sender, address(this), packetData.erc20Contract, packetData.amount);
-
         if (!packetData.originatorChainIsSource) {
+            // NOTE: Here we transfer the packetData.amount since the receiver chain is source that means that tokens
+            // that
+            // we are sending have arrived before from the receiving chain. The packetData.amount already took into
+            // consideration the conversion, since an ERC20 contract has been created with 6 decimals.
+            _transferFrom(sender, address(this), packetData.erc20Contract, packetData.amount);
             // receiver chain is source: burn the vouchers
             // TODO: Implement escrow balance tracking (#6)
             IBCERC20 ibcERC20Contract = IBCERC20(packetData.erc20Contract);
             ibcERC20Contract.burn(packetData.amount);
+        } else {
+            // We are sending out native EVM ERC20 tokens. We need to take into account the conversion
+            // NOTE: uint64 _sdkCoinAmount returned by SdkCoin._ERC20ToSdkCoin_ConvertAmount is discarded because it
+            // won't
+            // be used here. Recall that the _transferFrom function requires an uint256.
+            (, uint256 _remainder) = SdkCoin._ERC20ToSdkCoin_ConvertAmount(packetData.erc20Contract, packetData.amount);
```

Comment:
Why is the first argument ignored, shouldn't it be used? And why is `sendTransfer` function not modified?